core/src/main/java/org/infinispan/configuration/cache/IndexingConfigurationBuilder.java

```diff
@@ -118,8 +134,11 @@ public IndexingConfigurationBuilder withProperties(Properties props) {
 
    /**
     * Indicates indexing mode
+    * @deprecated This configuration can be removed as the index mode is calculated automatically.
     */
+   @Deprecated
    public IndexingConfigurationBuilder index(Index index) {
+      if (index != Index.NONE) enabled(true);
```

Comment:
The conditional thing seems wrong. This should probably be just ```enabled(index !=null && index != Index.NON);```

Reply:
ok, it is assuming nobody would pass a null to it, I will change